### ISSUE TITLE
Example ? Change text and parametrs

### ISSUE THREAD

**Issue by nurbat** [migueldeicaza/gui.cs#311]
void Update()
        {
            textView.Text = textNew += "demo \n";
            textView.Redraw(textView.Bounds);
            textView.SetNeedsDisplay();//.ChildNeedsDisplay();
// textView.Redraw(new Rect());
        }

` public void Awake()
        {
            Terminal.Gui.Application.Init();

            var top = Terminal.Gui.Application.Top;

            // Creates the top-level window to show
            var win = new Window("Server Menu")
            {
                X = 0,
                Y = 1, // Leave one row for the toplevel menu

                // By using Dim.Fill(), it will automatically resize without manual intervention
                Width = Dim.Fill(),
                Height = Dim.Fill(),
            };

            top.Add(win);
            
            // Creates a menubar, the item "New" has a help menu.
            var menu = new MenuBar(new MenuBarItem[] {
            new MenuBarItem ("_File", new MenuItem [] {
                new MenuItem ("_New", "Creates new file", null),
                new MenuItem ("_Close", "", Terminal.Gui.Application.RequestStop),
                new MenuItem ("_Quit", "", Application.Quit)
            }),
            new MenuBarItem ("_Edit", new MenuItem [] {
                new MenuItem ("_Copy", "", null),
                new MenuItem ("C_ut", "", null),
                new MenuItem ("_Paste", "", null)
            })
        });

            top.Add(menu);

            textView = new TextView()
            {
                Text = textNew,
                ReadOnly = true,
                Height = Dim.Percent(20),
            };
            top.Add(textView);

            var login = new Label($"Login: ") { X = 3, Y = 2 };
            var password = new Label("Password: ")
            {
                X = Pos.Left(login),
                Y = Pos.Top(login) + 1
            };
            var loginText = new TextField("")
            {
                X = Pos.Right(password),
                Y = Pos.Top(login),
                Width = 40
            };
            var passText = new TextField("")
            {
                Secret = true,
                X = Pos.Left(loginText),
                Y = Pos.Top(password),
                Width = Dim.Width(loginText)
            };

            // Add some controls, 
            win.Add(
                // The ones with my favorite layout system
                login, password, loginText, passText,

                    // The ones laid out like an australopithecus, with absolute positions:
                    new CheckBox(3, 6, "Remember me"),
                    new RadioGroup(3, 8, new[] { "_Personal", "_Company" }),
                    new Button(3, 14, "Ok"),
                    new Button(10, 14, "Cancel"),
                    new Label(3, 18, "Press F9 or ESC plus 9 to activate the menubar"));

            Terminal.Gui.Application.Run();
        }
`

How correct update UI ?

**Comment by nurbat** [comment]
![изображение](https://user-images.githubusercontent.com/5789051/69494957-35fa4c00-0eca-11ea-9b24-fe77970340f7.png)

**Comment by kbilsted** [comment]
Try setting `Height = 1` on all text fields. I'm creating a bugfix for that issue now

**Comment by BDisp** [comment]
@nurbat is this what you intend to do?

![serverchat](https://user-images.githubusercontent.com/13117724/69497130-73160c80-0ed1-11ea-8822-c661bec7caea.png)

`
	class ServerChat {
		TextView textView;
		private string textNew;

		public ServerChat ()
		{
			textNew = "";
			Awake ();
		}

		public void Awake ()
		{
			Terminal.Gui.Application.Init ();

			var top = Terminal.Gui.Application.Top;

			// Creates a menubar, the item "New" has a help menu.
			var menu = new MenuBar (new MenuBarItem [] {
				new MenuBarItem ("_File", new MenuItem [] {
				    new MenuItem ("_New", "Creates new file", null),
				    new MenuItem ("_Close", "", Terminal.Gui.Application.RequestStop),
				    new MenuItem ("_Quit", "", () => top.Running=false)
				}),
				new MenuBarItem ("_Edit", new MenuItem [] {
				    new MenuItem ("_Copy", "", null),
				    new MenuItem ("C_ut", "", null),
				    new MenuItem ("_Paste", "", null)
				})
			    });

			top.Add (menu);

			textView = new TextView () {
				Y = 1,
				Text = textNew,
				ReadOnly = true,
				Height = Dim.Percent (20),
			};
			top.Add (textView);

			// Creates the top-level window to show
			var win = new Window ("Server Menu") {
				X = 0,
				Y = Pos.Bottom(textView), // Leave one row for the toplevel menu

				// By using Dim.Fill(), it will automatically resize without manual intervention
				Width = Dim.Fill (),
				Height = Dim.Fill (),
			};

			var login = new Label ($"Login: ") { X = 3, Y = 2 };
			var password = new Label ("Password: ") {
				X = Pos.Left (login),
				Y = Pos.Top (login) + 1
			};
			var loginText = new TextField ("") {
				X = Pos.Right (password),
				Y = Pos.Top (login),
				Width = 40
			};
			var passText = new TextField ("") {
				Secret = true,
				X = Pos.Left (loginText),
				Y = Pos.Top (password),
				Width = Dim.Width (loginText)
			};

			Button button = new Button (3, 14, "OK");
			button.Clicked += () => { Update (); };
			// Add some controls, 
			win.Add (
			    // The ones with my favorite layout system
			    login, password, loginText, passText,

				// The ones laid out like an australopithecus, with absolute positions:
				new CheckBox (3, 6, "Remember me"),
				new RadioGroup (3, 8, new [] { "_Personal", "_Company" }),
				button,
				new Button (10, 14, "Cancel"),
				new Label (3, 18, "Press F9 or ESC plus 9 to activate the menubar"));

			top.Add (win);

			Terminal.Gui.Application.Run ();
		}

		void Update ()
		{
			textView.Text = textNew += "demo \n";
			textView.Redraw (textView.Bounds);
			textView.SetNeedsDisplay ();//.ChildNeedsDisplay();
						    // textView.Redraw(new Rect());
		}
	}

`

**Comment by nurbat** [comment]
Yes, but when you try to change the text update, the entire interface goes missing as in the screenshot above

**Comment by BDisp** [comment]
@nurbat the interface disappears because you misapplied it. Have you tried the code I submitted above? You can remove password.Text = password.Text == "Password:"? "Key:": "Password:"; because it was just for testing that changing the Label Text content updated properly.

**Comment by nurbat** [comment]
@BDisp Thanks, it works. I 'll try to add ScrollView to it now.

**Comment by nurbat** [comment]
```
            scrollView = new ScrollView(new Rect(0, 1, top.Frame.Width - 3, 3))
            {
                ShowVerticalScrollIndicator = true,
                ShowHorizontalScrollIndicator = false,
                X = 1,
                Y = 1,
                Width = Dim.Fill(),
                Height = Dim.Percent(20),
                ContentSize = new Size(160, 400),
                ContentOffset = new Point(0, 0),
            };

            top.Add(scrollView);

            textView = new TextView()
            {
                Y = 1,
                Text = textNew,
                ReadOnly = true,
                Height = Dim.Percent(20),
            };
            scrollView.Add(textView);
```

![изображение](https://user-images.githubusercontent.com/5789051/69557474-56e89d00-0faf-11ea-8f7b-835e2d2a4a73.png)

**Comment by nurbat** [comment]
![изображение](https://user-images.githubusercontent.com/5789051/69557825-e5f5b500-0faf-11ea-94bc-1344538aa493.png)

**Comment by BDisp** [comment]
It's not good. So you are always rebuilding the UI and it turns out to be irresponsive
